### DISCUSSION TITLE
Problem with GENConv on bipartite  graph

### DISCUSSION THREAD

**Discussion by abdalgader-a-zz** [pyg-team/pytorch_geometric#5861] [Q&A] [answered]
Hi, 

Does GENConv in pyg 2.0.4 supporting bipartite graph? 
It says supported in [GNN Cheatsheet](https://pytorch-geometric.readthedocs.io/en/2.0.4/notes/cheatsheet.html#:~:text=%E2%9C%93-,GENConv%20(Paper),-%E2%9C%93)

I got these errors on  testing cases:

> from torch_geometric.nn import GENConv
> GENConv(-1, 128)

`RuntimeError: Trying to create tensor with negative dimension -2: [-2]`

and, 

> from torch_geometric.nn import GENConv
> GENConv((16, 16), 128)

`TypeError: '>' not supported between instances of 'tuple' and 'int'`

do I need any updgrade?

**Comment by EdisonLeeeee** [comment, marked as answer]
Yes, this has been resolved in master and you need to upgrade your PyG as described [here](https://github.com/pyg-team/pytorch_geometric#nightly-and-master).

**Reply by abdalgader-a-zz** [reply]
Thanks! upgrading solve the issue.
